### PR TITLE
feat(secrets): Skip invalid secrets checks + soft/hard fails

### PR DESCRIPTION
**Pull request by Eliran-Turgeman**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
* Allowing usage of `--skip-check invalid` to suppress all invalid secrets findings from secrets report
* Adding hard/soft-fail-on for all secrets validation statuses
* Minor tests adjustements
*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/runner_filter.py [hunk 1]
```diff
@@ -262,6 +264,10 @@ def within_threshold(self, severity: Severity) -> bool:
         below_max = self.skip_check_threshold and self.skip_check_threshold.level >= severity.level
         return above_min and not below_max
 
+    @staticmethod
+    def secret_check_matches(secret_validation_status: str, statuses_list: list[str]) -> bool:
+        return secret_validation_status.lower() in [status.lower() for status in statuses_list]
```
Comment (maxamel) [inline checkov/runner_filter.py:269]: Might be worth having a list of lower case statuses so we don't iterate each time to create the lower case list, then iterate again to find the secret_validation_status, WDYT?

File: checkov/runner_filter.py [hunk 2]
```diff
@@ -262,6 +264,10 @@ def within_threshold(self, severity: Severity) -> bool:
         below_max = self.skip_check_threshold and self.skip_check_threshold.level >= severity.level
         return above_min and not below_max
 
+    @staticmethod
+    def secret_check_matches(secret_validation_status: str, statuses_list: list[str]) -> bool:
```
Comment (marynaKK) [inline checkov/runner_filter.py:268]: Is check_matches supposed to run only after the status of the secret is validated? 
If so - maybe the name can reference this, as `post_secret_check_matches ` or something like this.
I initially thought this is a similar method to `check_matches` but for `secrets`

File: tests/secrets/test_secrets_verification.py [hunk 3]
```diff
@@ -11,26 +12,24 @@
 from checkov.secrets.runner import Runner
 
 
+@mock.patch.dict(os.environ, {"CKV_VALIDATE_SECRETS": "true"})
```
Comment (marynaKK) [inline tests/secrets/test_secrets_verification.py:15]: NICCE 🍒

File: tests/secrets/test_secrets_verification.py [hunk 4]
```diff
@@ -45,6 +44,7 @@ def test_verify_secrets_insufficient_params_no_flag() -> None:
 
 
 @responses.activate
+@mock.patch.dict(os.environ, {"CKV_VALIDATE_SECRETS": "true"})
```
Comment (marynaKK) [inline tests/secrets/test_secrets_verification.py:47]: why remove here the `BC_API_KEY` ? :)

File: tests/secrets/test_secrets_verification.py [hunk 5]
```diff
@@ -69,9 +66,8 @@ def test_verify_secrets_failure(mock_bc_integration, status_code: int) -> None:
...
```
Comment (marynaKK) [inline tests/secrets/test_secrets_verification.py:69]: also here, why remove `BC_API_KEY`?

**Review by marynaKK — COMMENTED** [review]
:) good job! had few questions/suggestions

File: checkov/runner_filter.py (hunk 1 above)
Comment (Eliran-Turgeman) [reply checkov/runner_filter.py]: When adding the soft/hard fails I changed it to insert the values capitalized if its a secret validation status + added some tests. removed the .lower() from both objects

File: tests/secrets/test_secrets_verification.py (hunk 4 above)
Comment (Eliran-Turgeman) [reply tests/secrets/test_secrets_verification.py:47]: It is an unnecessary key, the condition to run secrets verify is on `bc_integration.bc_api_key` and not on the environment variable

**Review by maxamel — APPROVED** [review]

File: checkov/runner_filter.py (hunk 2 above)
Comment (Eliran-Turgeman) [reply checkov/runner_filter.py:271]: changed to `secret_validation_status_matches`, wdyt?

**Review by marynaKK — APPROVED** [review]

**Pull request merged**